scripts/polygon/deploy.js

```diff
@@ -0,0 +1,189 @@
+const { BigNumber } = require("@ethersproject/bignumber");
+const hre = require("hardhat");
+const ethers = hre.ethers;
+
+const deployPickleToken = async () => {
+  console.log("deploying pickle token...");
+
+  const childChainManagerProxy = "0xA6FA4fB5f76172d178d61B04b0ecd319C5d1C0aa";
+  const minter = "0xaCfE4511CE883C14c4eA40563F176C3C09b4c47C";
+
+  const PickleTokenFactory = await ethers.getContractFactory(
+    "src/flatten/pickle-token.sol:PickleToken"
+  );
+  const PickleToken = await PickleTokenFactory.deploy(
+    "PickleToken",
+    "PICKLE",
+    18,
+    childChainManagerProxy,
+    minter
+  );
+  console.log("pickle token deployed at ", PickleToken.address);
+};
+
+const deployMasterChef = async () => {
+  console.log("deploying masterchef...");
+
+  const pickle = "0x6c551cAF1099b08993fFDB5247BE74bE39741B82";
+  const devaddr = "0xaCfE4511CE883C14c4eA40563F176C3C09b4c47C";
+  const picklePerBlock = 1000000000000;
+  const startBlock = 13620000;
+  const bonusEndBlock = 0;
+
+  const MasterChefFactory = await ethers.getContractFactory(
+    "src/flatten/masterchef.sol:MasterChef"
```

Comment:
why did you need to flatten the contracts?

Reply:
we need to deploy flatten code and verify using flatten code. It doesn't support multi-file choice as well.